extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -113,18 +114,26 @@ void unremovable(Capabilities capabilities, BuildProducer<UnremovableBeanBuildIt
 
     @BuildStep
     void register(
+            CurateOutcomeBuildItem curateOutcomeBuildItem,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchyClass,
             BuildProducer<ReflectiveMethodBuildItem> reflectiveMethod,
             BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
         reflectiveClass.produce(
-                new ReflectiveClassBuildItem(true, false, "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector",
+                new ReflectiveClassBuildItem(true, false,
                         "com.fasterxml.jackson.databind.ser.std.SqlDateSerializer",
                         "com.fasterxml.jackson.databind.ser.std.SqlTimeSerializer",
                         "com.fasterxml.jackson.databind.deser.std.DateDeserializers$SqlDateDeserializer",
                         "com.fasterxml.jackson.databind.deser.std.DateDeserializers$TimestampDeserializer",
                         "com.fasterxml.jackson.annotation.SimpleObjectIdResolver"));
 
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
+                x -> x.getGroupId().equals("com.fasterxml.jackson.module")
+                        && x.getArtifactId().equals("jackson-module-jaxb-annotations"))) {
```

Comment:
I know I've asked elsewhere too so apologies for that, but did we reach a consensus about this pattern?

Reply:
It looks like there is no better way at the moment (see https://github.com/quarkusio/quarkus/pull/29886#discussion_r1054703116)

Reply:
Thanks